samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManagerFactory.scala

```diff
@@ -91,8 +99,10 @@ class KafkaCheckpointManagerFactory extends CheckpointManagerFactory with Loggin
       socketTimeout,
       consumerConfig.socketReceiveBufferBytes,
       consumerConfig.fetchMessageMaxBytes,            // must be > buffer size
+      systemConsumerF,
+      systemAdminF,
       new ClientUtilTopicMetadataStore(producerConfig.bootsrapServers, clientId, socketTimeout),
-      connectProducer,
+      systemProducerF,
```

Comment:
Why do we take in functions for `systemAdminF`, `systemProducerF` and `systemConsumerF`. Is n't what the `SystemFactory` is doing with its methods for `createProducer`, `createConsumer` and `createAdmin`? It might be simpler to take in the `SystemFactory` right?

Reply:
that's kind of bigger refactoring - if you pass the systemFactory, then you also need to pass config. If you pass the config, then you also can avoid passing multiple/most of the other arguments and derive them from the config inside the method.